metricbeat/module/mssql/io/data.go

```diff
@@ -0,0 +1,80 @@
+package io
+
+import (
+	s "github.com/elastic/beats/libbeat/common/schema"
+	c "github.com/elastic/beats/libbeat/common/schema/mapstrstr"
+	"time"
+)
+
+var (
+	schema = s.Schema{
+		//Returns the list of tape devices and the status of mount requests for backups.
+		"backup_tapes": s.Object{
+			"physical_device_name":    c.Str("physical_device_name"),
+			"logical_device_name":     c.Str("logical_device_name"),
+			"status":                  c.Int("status"),
+			"status_desc":             c.Str("status_desc"),
+			"mount_request_time":      c.Time(time.RFC3339, "mount_request_time"),
+			"mount_expiration_time":   c.Time(time.RFC3339, "mount_expiration_time"),
+			"database_name":           c.Str("database_name"),
+			"spid":                    c.Int("spid"),
+			"command":                 c.Int("command"),
+			"command_desc":            c.Str("command_desc"),
+			"media_family_id":         c.Int("media_family_id"),
+			"media_set_name":          c.Str("media_set_name"),
+			"media_set_guid":          c.Str("media_set_guid"),
+			"media_sequence_number":   c.Int("media_sequence_number"),
+			"tape_operation":          c.Int("tape_operation"),
+			"tape_operation_desc":     c.Str("tape_operation_desc"),
+			"mount_request_type":      c.Int("mount_request_type"),
+			"mount_request_type_desc": c.Str("mount_request_type_desc"),
+		},
+		//This view returns the drive name of each of the shared drives if the current server instance is a
+		// clustered server. If the current server instance is not a clustered instance it returns an empty rowset.
+		"cluster_shared_drives": s.Object{
+			"drive_name":  c.Str("DriveName"),
+			"pdw_node_id": c.Int("pdw_node_id"),
+		},
+		//Returns a row for each pending I/O request in SQL Server.
```

Comment:
If there is a different row for each request, then it'd be better to move them to their own metricset, and generate an event for each one of them.

Here you can keep a summary in any case, with the count of pending requests, total pending ticks...

Reply:
I haven't done this yet on the current commit. I need to investigate a bit more how it works :) but I think I understand the overall idea